libr/core/vmenus.c

```diff
@@ -4061,8 +4073,26 @@ R_API void r_core_visual_define(RCore *core, const char *args, int distance) {
 		ut64 try_off;
 		bool found = false;
 		for (try_off = start_off; try_off < start_off + incr*16; try_off += incr) {
-			int ret = r_anal_op (core->anal, &op, try_off,
-				core->block + try_off - core->offset, 32, R_ANAL_OP_MASK_ALL);
+            
+			// Prepares jump to segfaulty r_anal_op function
+			struct sigaction sa;
+			memset(&sa, 0, sizeof(sigaction));
+			sigemptyset(&sa.sa_mask);
+			sa.sa_flags = SA_NODEFER;
+			sa.sa_sigaction = segfaulty_action_handler;
+			sigaction(SIGSEGV, &sa, NULL);
+			
+			int ret;
+			if (setjmp(point) == 0)
+			{
+			    // Call segfaulty function
+			    ret = r_anal_op (core->anal, &op, try_off, core->block + try_off - core->offset, 32, R_ANAL_OP_MASK_ALL);
```

Comment:
the problem is when try_off - core->offset is < 0 or > core->blocksize, using r_core_anal_op() should be fine because we just need to provide the address and chk for null instead